src/ProjectTemplates/Web.ProjectTemplates/content/SignalR-CSharp/Hubs/StreamingTimerHub.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Company.WebApplication1
+{
+    public class StreamingTimerHub : Hub
+    {
+        public ChannelReader<string> ServerTimer(CancellationToken token)
+        {
+            var channel = Channel.CreateUnbounded<string>();
+            _ = WriteDateAsync(channel.Writer, token);
+            return channel.Reader;
+        }
+
+        private async Task WriteDateAsync(ChannelWriter<string> writer, 
+            CancellationToken token)
+        {
+            try
+            {
+                while(true)
+                {
+                    token.ThrowIfCancellationRequested();
+                    await writer.WriteAsync(DateTime.Now.ToString("HH:mm:ss.fffffff"));
+                    await Task.Delay(10, token);
```

Comment:
Is 10ms a good time for the timer? That's like a *lot* of ticks, isn't it?

Reply:
Also, fun fact, on Windows `Task.Delay` will be ~15ms for any value between 1 and 15 because of OS limitations.

Reply:
this was designed to show the streaming stuff "goes real fast." per @Tratcher i will be omitting the streaming part.